### PR TITLE
bug-fix/PLUGIN-94

### PR DESCRIPTION
**Pull request by ailegion**
fixed: PLUGIN-94 bumped spark version to 2.3.0

JIRA: https://cdap.atlassian.net/browse/PLUGIN-94

Did the following test:
  * Bump the Spark version on pom to 2.3.0
  * Test Kafka streaming source
  * Pipeline worked successfully

The tests were done on Sandbox

### CODE REVIEW

File: kafka-plugins-0.10/pom.xml [hunk 1]
```diff
@@ -48,6 +52,10 @@
           <groupId>org.apache.spark</groupId>
           <artifactId>spark-tags_2.11</artifactId>
         </exclusion>
+        <exclusion>
+          <groupId>net.jpountz.lz4</groupId>
```
Comment (CuriousVini) [inline kafka-plugins-0.10/pom.xml:56]: why is this exclusion needed?

**Review by CuriousVini — COMMENTED** [review]
1. Please test these changes on dataproc
2. The package contains 3 plugins - kafka realtime source, kafka realtime sink and kafka batch source
3. backwards compatibility - does it work on 2.2.0 pipelines after this change?

File: kafka-plugins-0.10/pom.xml (hunk 1 above)
Comment (ardian4) [reply kafka-plugins-0.10/pom.xml:56]: @CuriousVini  without exclusion we hit this exception:

`java.util.concurrent.ExecutionException: java.lang.NoSuchMethodError: net.jpountz.lz4.LZ4BlockInputStream.<init>(Ljava/io/InputStream;Z)V
        at com.google.common.util.concurrent.AbstractFuture$Sync.getValue(AbstractFuture.java:306) ~[hive-exec-1.2.1.jar:1.2.1]
        at com.google.common.util.concurrent.AbstractFuture$Sync.get(AbstractFuture.java:293) ~[hive-exec-1.2.1.jar:1.2.1]
        at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:116) ~[hive-exec-1.2.1.jar:1.2.1]
        at io.cdap.cdap.app.runtime.spark.SparkRuntimeService.run(SparkRuntimeService.java:346) ~[cdap-spark-core2_2.11-6.4.0-SNAPSHOT.jar:na]
        at com.google.common.util.concurrent.AbstractExecutionThreadService$1$1.run(AbstractExecutionThreadService.java:52) ~[hive-exec-1.2.1.jar:1.2.1]
        at io.cdap.cdap.app.runtime.spark.SparkRuntimeService$5$1.run(SparkRuntimeService.java:404) [cdap-spark-core2_2.11-6.4.0-SNAPSHOT.jar:na]
        at java.lang.Thread.run(Thread.java:748) [na:1.8.0_181]
Caused by: java.lang.NoSuchMethodError: net.jpountz.lz4.LZ4BlockInputStream.<init>(Ljava/io/InputStream;Z)V
        at org.apache.spark.io.LZ4CompressionCodec.compressedInputStream(CompressionCodec.scala:122) ~[spark-core_2.11-2.3.0.jar:2.3.0]
        at org.apache.spark.streaming.Checkpoint$$anonfun$deserialize$2.apply(Checkpoint.scala:169) ~[spark-streaming_2.11-2.3.0.jar:2.3.0]
        at org.apache.spark.streaming.Checkpoint$$anonfun$deserialize$2.apply(Checkpoint.scala:162) ~[spark-streaming_2.11-2.3.0.jar:2.3.0]
        at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1377) ~[spark-core_2.11-2.3.0.jar:2.3.0]
        at org.apache.spark.streaming.Checkpoint$.deserialize(Checkpoint.scala:175) ~[spark-streaming_2.11-2.3.0.jar:2.3.0]
        at org.apache.spark.streaming.CheckpointReader$$anonfun$read$2.apply(Checkpoint.scala:356) ~[spark-streaming_2.11-2.3.0.jar:2.3.0]`

Comment (CuriousVini) [reply kafka-plugins-0.10/pom.xml:56]: I see. That seems like a class conflict. Does it only happen for 2.3 spark version and not for 2.2? 
Also could you please share results for `mvn dependency:tree` without this exclusion via email?

**Comment by CuriousVini** [comment]
@bajram-adapt @ailegion Has this change been verified on dataproc?

**Comment by ailegion** [comment]
> @bajram-adapt @ailegion Has this change been verified on dataproc?

tested on 6.2 and it works: 
image

**Comment by CuriousVini** [comment]
@ailegion what version of spark was this change tested on? We should make sure there is no regression. So good to test on 2.2 and 2.3
https://cloud.google.com/dataproc/docs/concepts/versioning/dataproc-versions

You can specify the dataproc image container 2.2 and 2.3 spark versions in runtime argument: 
`system.profile.properties.imageVersion` - `dataproc_image`

**Comment by bajram-adapt** [comment]
@CuriousVini created Jira item for spark 2.2 pipeline failure discussed on Friday: https://cdap.atlassian.net/browse/PLUGIN-586

**Review by CuriousVini — COMMENTED** [review]
Please squash the commits

**Comment by ailegion** [comment]
> Please squash the commits

squashed

**Pull request merged**